src/main/kotlin/org/rust/ide/newProject/ui/AddUserTemplateDialog.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.newProject.ui
+
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.ui.components.JBTextField
+import com.intellij.ui.layout.panel
+import org.rust.ide.newProject.state.RsUserTemplate
+import org.rust.ide.newProject.state.RsUserTemplatesState
+import java.awt.Dimension
+import javax.swing.JComponent
+import javax.swing.event.DocumentEvent
+import javax.swing.event.DocumentListener
+
+class AddUserTemplateDialog : DialogWrapper(null) {
+    private val repoLinkField: JBTextField = JBTextField().apply {
+        preferredSize = Dimension(400, 0)
+        document.addDocumentListener(object : DocumentListener {
+            override fun insertUpdate(e: DocumentEvent?) {
+                if (nameField.text.isNotBlank()) return
```

Comment:
Let's extract this code to a method like `private fun suggestName()` to avoid `this@apply` usages